narwhal/network/src/connectivity.rs

```diff
@@ -115,6 +141,87 @@ impl ConnectionMonitor {
 
         self.connection_statuses.insert(peer_id, connection_status);
     }
+
+    // TODO: Replace this with ClosureMetric
+    fn update_quinn_metrics_for_peer(&self, peer_id: &PeerId, stats: &ConnectionStats) {
+        // Update PathStats
+        self.connection_metrics
+            .network_peer_rtt
+            .with_label_values(&[&format!("{peer_id}")])
```

Comment:
Also, you should create a single "peer_id" string and reuse it everywhere so that you're allocating 1 time and not N times.